### PR TITLE
Adding code quality tools

### LINKED ISSUE #1: Quality Control tasks
Quality Control items to set up:

* File templates
* Project code style
* Lint settings
* Checkstyle settings
* PMD settings
* Findbugs settings
* Jacoco coverage reporting
* ...

We can tie a lot of these tools together into the 'check' task so that contributors can check their code before submitting a pull request.

### PR DESCRIPTION
**Pull request by setheclark**
Added checkstyle/pmd/findbugs tasks to the project.  Just run the 'check' task. resolves #1

### CODE REVIEW

**Comment by antoinecampbell** [comment]
Not sure if the entire .idea folder was omitted, but can we share a code formatter for the project so we can automatically adhere to the styling?

**Comment by setheclark** [comment]
Yeah.  I can do that.  I'll just create it based off of Square's code style. Only thing with that is people have to set their code style to use the project's rather than  their own. Haven't found a way to make that automatic.

**Comment by setheclark** [comment]
Also, would we want to correct the project all at once or just fix it as we go?

**Comment by antoinecampbell** [comment]
I would think that would be better as a single commit, so future changes don't appear that an entire file changed.

File: config/quality/findbugs/android-exclude-filter.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        <Class name="~.*\.Manifest\$.*"/>
+    </Match>
+    <!-- All bugs in test classes, except for JUnit-specific bugs -->
+    <Match>
+        <Class name="~.*\.*Test" />
+        <Not>
+            <Bug code="IJU" />
+        </Not>
+    </Match>
+
```
Comment (OllenDev) [inline config/quality/findbugs/android-exclude-filter.xml]: extra spaces

**Pull request merged**